modules/BuildUtilities/src/server/python/zzz-config-to-envs.py

```diff
@@ -0,0 +1,146 @@
+        out_lines.append(out_line)
+    return(out_lines, name_maps)
+
+def process_configs(args):
+    out_lines = []
+    name_maps = {}
+    [example_lines, example_name_maps] = process_property_lines(open("conf/config.properties.example").readlines())
+    if args.properties:
+        lines = args.properties.readlines()
+
+        # lines = get_text_file_lines("conf/config.properties.example")
+        [out_lines, name_maps] = process_property_lines(lines, list(example_name_maps.keys()))
+
+    if args.cmpdreg:
+        # out_lines.append("\n# CmpReg settings\n")
+        cmpdreg_json = json.load(args.cmpdreg)
+        df = pd.json_normalize(cmpdreg_json, sep='.').add_prefix('client.cmpdreg.')
+        cmpdreg_configs = df.to_dict(orient='records')[0]
+        for key in cmpdreg_configs:
+            env_name = key.replace('.','_').upper()
+            setting = cmpdreg_configs[key]
+            if isinstance (setting, bool):
+                setting = json.dumps(setting)
+
+            name_maps[key] = {
+                            "env_name": env_name,
+                            "setting": str(setting)+"\n"
+            }
+            # out_line = f"{key}=${{env.{env_name}}}\n"
+            # out_lines.append(out_line)
+
+
+    env_file_lines = []
+    env_file_lines.append("ACAS_TAG=release-1.13.6\n\n")
+    for original_setting in name_maps:
+        for replace_setting in name_maps:
+            # If the setting inherits from another setting we need to replace the inherited settings with the
+            # actual setting becuase .env files don't support inheritance
+            name_maps[original_setting]['setting'] = os.path.expandvars(os.path.expandvars(name_maps[original_setting]['setting'].replace("env.","").replace(replace_setting, name_maps[replace_setting]['env_name'])))
+
+        # Format the environment variable line
+        example_value = None
+        if original_setting in example_name_maps:
+            example_value = example_name_maps[original_setting]['interpolated_value']
+        original_value = name_maps[original_setting]['setting']
+
+        compare_value = original_value.rstrip("\n")
+        if compare_value != example_value:
+            env_line = f"# MAPS TO: {original_setting}\n"
+            env_line = env_line+f"{name_maps[original_setting]['env_name']}={name_maps[original_setting]['setting']}\n"
+            env_file_lines.append(env_line)
+        else:
+            print(f"Skipping {original_setting} because it's value '{compare_value}' equals the default value '{example_value}'")
+
+    env_file_name = "new-environment.env"
+    write_lines_to_file(env_file_name, env_file_lines)
+    print(f"Wrote environment variables to {env_file_name}")
+    properties_file_name = "new-config.properties"
+    write_lines_to_file(properties_file_name, out_lines)
+    print(f"Wrote properties to {properties_file_name}")
+
+
+argparser = argparse.ArgumentParser(description='Process configs')
```

Comment:
Could use a more verbose explanation of what this script produces. It seems to produce two files: `new-config.properties` which lists only novel config files not found in `config.properties.example` and a `new-environment.env` which lists only override values which don't match the default values.

But I'm confused about how / where it's supposed to be invoked.